prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -0,0 +1,80 @@
+"""
+PYTHONPATH=. streamlit run prediction_market_agent/agents/microchain_agent/app.py
+
+Tip: if you specify PYTHONPATH=., streamlit will watch for the changes in all files, isntead of just this one.
+"""
+import streamlit as st
+
+from prediction_market_agent.utils import APIKeys, streamlit_asyncio_event_loop_hack
+
+# Imports using asyncio (in this case mech_client) cause issues with Streamlit
+streamlit_asyncio_event_loop_hack()
+
+from microchain import Agent
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.tools.costs import openai_costs
+
+from prediction_market_agent.agents.microchain_agent.microchain_agent import get_agent
+
+
+def check_api_keys() -> None:
+    keys = APIKeys()
+    if not keys.OPENAI_API_KEY:
+        st.error("No OpenAI API Key provided via env var/secret.")
+        st.stop()
+    elif not keys.BET_FROM_PRIVATE_KEY:
+        st.error("No wallet private key provided via env var/secret.")
+        st.stop()
```

Comment:
This is handy. We could even have something like `def check_all_required_keys(list_of_keys_that_are_required_for_given_app)`. But that's fine for another PR, I'm just commenting on an idea here.

Reply:
Good idea. I've made a new file, streamlit_utils.py and put `check_required_api_keys` in there (and moved `streamlit_asyncio_event_loop_hack` alongside it)